applications/IgaApplication/custom_elements/embedded_isogeometric_beam_element.cpp

```diff
@@ -0,0 +1,5629 @@
+//  KRATOS  _____________
+//         /  _/ ____/   |
+//         / // / __/ /| |
+//       _/ // /_/ / ___ |
+//      /___/\____/_/  |_| Application
+
+// System includes
+// External includes
+// Project includes
+#include "custom_elements/embedded_isogeometric_beam_element.h"
+#include <numeric>
+#include "utilities/math_utils.h"
+#include "geometries/nurbs_curve_geometry.h"
+
+//debugging only
+#include <sstream>
+#include <iomanip>
+
+namespace Kratos {
+    ///@name Degrees of freedom
+    ///@{
+
+    void EmbeddedIsogeometricBeamElement::EquationIdVector(
+        EquationIdVectorType& rResult,
+        const ProcessInfo& rCurrentProcessInfo
+    ) const
+    {
+        //KRATOS_WATCH("EmbeddedIsogeometricBeamElement::EquationIdVector");
+        KRATOS_TRY;
+
+        const auto& r_geometry = GetGeometry();
+        const SizeType number_of_control_points = r_geometry.size();
+
+        if (rResult.size() != 4 * number_of_control_points)
+            rResult.resize(4 * number_of_control_points, false);
+
+        const IndexType pos = r_geometry[0].GetDofPosition(DISPLACEMENT_X);
+
+        for (IndexType i = 0; i < number_of_control_points; ++i) {
+            const IndexType index = i * 4;
+            rResult[index] = r_geometry[i].GetDof(DISPLACEMENT_X, pos).EquationId();
+            rResult[index + 1] = r_geometry[i].GetDof(DISPLACEMENT_Y, pos + 1).EquationId();
+            rResult[index + 2] = r_geometry[i].GetDof(DISPLACEMENT_Z, pos + 2).EquationId();
+            rResult[index + 3] = r_geometry[i].GetDof(ROTATION_X, pos + 3).EquationId();
```

Comment:
just for curiosity, why do you use only rotation_X? don't you have the others?

Reply:
2d beams shpuld be in the xy plane...in that case it should be rotationz the relevant rotation

Reply:
hmm I think it's a 3D beam, it has 3 displ